version: "3"

networks:
  covidapp-network:
    driver: bridge
  [TRAEFIK_NETWORK]:
    external: true

services: 
  covid_app:
    image: kr/covidapp
    container_name: covidapp
    restart: unless-stopped
    networks:
      - covidapp-network
      - [TRAEFIK_NETWORK]
    build: .
    depends_on:
      - mysql
    labels:
      - traefik.http.routers.covidapp.rule=Host(`[SUBDOMAIN_URL]`)
      - traefik.http.routers.covidapp.tls=true
      - traefik.http.routers.covidapp.tls.certresolver=lets-encrypt
      - traefik.http.services.covidapp.loadbalancer.server.port=1337
  
  mysql:
      image: mysql/mysql-server:latest
      container_name: mysql_covidapp
      restart: unless-stopped
      environment:
        MYSQL_ROOT_PASSWORD: '[MYSQL_ROOT_PASS]'
      command: mysqld --sql_mode="" --character-set-server=utf8mb4 --init-connect='SET NAMES UTF8;' --innodb-flush-log-at-trx-commit=0
      volumes:
        - /pwspool/software/covid19/mysql:/var/lib/mysql
        - ./schema:/docker-entrypoint-initdb.d/:ro
      networks:
        - covidapp-network
      labels:
        - traefik.enable=false

